### ISSUE TITLE
[Bug] viewDays switches on window resize

### ISSUE THREAD

**Issue by jaruszewskiG** [mounthorse-slns/angular-calendar-scheduler#47]
Hi, I've encountered a problem. When I resize the browser's window, the viewDays setting changes by itself. Video below shows it all: https://drive.google.com/file/d/1NDLKCwTxVBjg0ARSXf4aZfenzgUr9jnf/view?usp=sharing

Expected behaviour: viewDays doesn't change on browser/calendar width change.

TS:
```
import { Component, OnInit, Inject, LOCALE_ID } from '@angular/core';
import { Subject } from 'rxjs';

import {
  endOfDay,
  addMonths,
  isToday
} from 'date-fns';
import {
  SchedulerViewHourSegment,
  CalendarSchedulerEvent,
  CalendarSchedulerEventAction,
  SchedulerDateFormatter,
  SchedulerEventTimesChangedEvent
} from 'angular-calendar-scheduler';
import {
  CalendarView,
  CalendarDateFormatter,
} from 'angular-calendar';
import { CalendarService } from '@services';

@Component({
  selector: 'app-calendar-wrapper',
  templateUrl: './calendar-wrapper.component.html',
  styleUrls: ['./calendar-wrapper.component.scss'],
  providers: [{
    provide: CalendarDateFormatter,
    useClass: SchedulerDateFormatter
  }]
})
export class CalendarWrapperComponent implements OnInit {
  public viewDate: Date = new Date();
  public refresh: Subject<any> = new Subject();
  public locale = 'en';
  public dayStartHour = new Date().getHours();
  public dayEndHour = 22;
  public events: CalendarSchedulerEvent[];
  public segmentModifier: () => void;
  public eventModifier: () => void;
  // eslint-disable-next-line @typescript-eslint/naming-convention
  public get CalendarView() {
    return CalendarView;
  }

  private minDate: Date = new Date();
  private maxDate: Date = endOfDay(addMonths(new Date(), 1));
  private actions: CalendarSchedulerEventAction[] = [
    {
      when: 'enabled',
      label: '<span class="valign-center"><i class="material-icons md-18 md-red-500">cancel</i></span>',
      title: 'Delete',
      onClick: (event: CalendarSchedulerEvent): void => {
        console.log('Pressed action \'Delete\' on event ' + event.id);
      }
    },
    {
      when: 'cancelled',
      label: '<span class="valign-center"><i class="material-icons md-18 md-red-500">autorenew</i></span>',
      title: 'Restore',
      onClick: (event: CalendarSchedulerEvent): void => {
        console.log('Pressed action \'Restore\' on event ' + event.id);
      }
    }
  ];

  constructor(
    private calendarService: CalendarService,
    @Inject(LOCALE_ID) locale: string,
  ) {
    this.locale = locale;

    this.segmentModifier = ((segment: SchedulerViewHourSegment): void => {
      segment.isDisabled = !this.isDateValid(segment.date);
    }).bind(this);

    this.eventModifier = ((event: CalendarSchedulerEvent): void => {
      event.isDisabled = !this.isDateValid(event.start);
    }).bind(this);
  }

  ngOnInit(): void {
    this.calendarService.getEvents(this.actions)
      .then((events: CalendarSchedulerEvent[]) => this.events = events);
  }

  segmentClicked(action: string, segment: SchedulerViewHourSegment): void {
    // TODO 22.01.2020: Remove console.logs and implement handling segment clicked event
    console.log('segmentClicked Action', action);
    console.log('segmentClicked Segment', segment);
    // /TODO
  }

  eventClicked(action: string, event: CalendarSchedulerEvent): void {
    // TODO 22.01.2020: Remove console.logs and implement handling event clicked event
    console.log('eventClicked Action', action);
    console.log('eventClicked Event', event);
    // /TODO
  }

  eventTimesChanged({ event, newStart, newEnd, type }: SchedulerEventTimesChangedEvent): void {
    // TODO 22.01.2020: Replace with actual implementation of handling event times changed event
    console.log('eventTimesChanged Type', type);
    console.log('eventTimesChanged Event', event);
    console.log('eventTimesChanged New Times', newStart, newEnd);
    const ev: CalendarSchedulerEvent = this.events.find(e => e.id === event.id);
    ev.start = newStart;
    ev.end = newEnd;
    this.refresh.next();
    // /TODO
  }

  private isDateValid(date: Date): boolean {
    return isToday(date) || date >= this.minDate && date <= this.maxDate;
  }
}

```

HTML:
```
<mat-progress-spinner *ngIf="!events" fxLayout="row" fxLayoutAlign="center center" color="primary" mode="indeterminate" class="mx-auto mt-5">
</mat-progress-spinner>
<div class="calendar-container" *ngIf="events">
  <calendar-scheduler-view
    [viewDate]="viewDate"
    [events]="events"
    [locale]="locale"
    [forceViewDays]="1"
    [weekStartsOn]="1"
    [startsWithToday]="true"
    [hourSegments]="4"
    [dayStartHour]="dayStartHour"
    [dayEndHour]="dayEndHour"
    [dayModifier]="dayModifier"
    [hourModifier]="hourModifier"
    [segmentModifier]="segmentModifier"
    [eventModifier]="eventModifier"
    [showEventActions]="true"
    [zoomEventOnHover]="true"
    [refresh]="refresh"
    (dayHeaderClicked)="dayHeaderClicked($event.day)"
    (hourClicked)="hourClicked($event.hour)"
    (segmentClicked)="segmentClicked('Clicked', $event.segment)"
    (eventClicked)="eventClicked('Clicked', $event.event)"
    (eventTimesChanged)="eventTimesChanged($event)">
  </calendar-scheduler-view>
</div>
```

**Comment by mounthorse-slns** [comment]
Hi @jaruszewskiG .
I've just release a new version to fix this behaviour.
The `forceViewDays` input has been removed. Now to specify the number of days shown you have to call a new method on the component `setViewDays(viewDays: number)` (taking a reference to the component using `ViewChild`)

Moreover, I've added a new boolean input `responsive`.
If set to false the calendar doesn't change on browser/calendar width change anymore (default value is `false`).

You can find an example in the README too.

I hope this reseove your issue. Let me know ;)

**Comment by jaruszewskiG** [comment]
@mounthorse-slns , thanks for the quick fix, it works great, but now when you use the `setViewDays` method, you're getting `Error NG0100: ExpressionChangedAfterItHasBeenCheckedError` (or at least I do ;)). Can you please check if it also occurs for you? I'm calling the method exactly like you did in the README, by reference, using ViewChild. Calling `.next` on `refresh` after calling `setViewDays` method doesn't help sadly.

See the error below:
https://i.imgur.com/wTyLoQ5.png

One more thing, I want the component to show one day view on load, so now I'm calling the `setViewDays` method in the `ngDoCheck` hook. Is there a better way to achieve this? See the code snippet below:
```
  ngDoCheck(): void {
    if (this.calendarScheduler?.calendarId) {
      this.calendarScheduler.setViewDays(1);
    }
  }
```

**Comment by mounthorse-slns** [comment]
@jaruszewskiG I've published the version `10.1.3`. Now there is a new input `viewDays` that allow you to set the initial number of days to show.
Check it out and let me know ;)

**Comment by jaruszewskiG** [comment]
@mounthorse-slns wow, that's a quick response! Works like a charm, thank you! :)
